### PR TITLE
Connect the home page view to the DB (for loading updates)

### LINKED ISSUE #11: Finish Home Page Content Conversion to DB Driven
See how the National category is built on the home page. 

Write a migration and update the view to be fully db driven.

### PR DESCRIPTION
**Pull request by NimaBoscarino**
Fixes #11 

This PR refactors the Home view a bit, and loads the daily_update resources (and categories) from the DB. I added all the current resources to the seeds, so it should be possible to just run `rake db:seed` once this is merged.

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class Category < ApplicationRecord
 
-  has_many :resources
+  has_many :resources, dependent: :destroy
```
Comment (garrettgsb) [inline app/models/category.rb]: Are you sure we want to destroy resources if they're associated with a deleted category? I'd assume being uncategorized would be fine, but lmk if I'm missing something.

File: app/views/pages/_resource-card.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    <div class="card-body">
+    <h4 class="card-title"><%= resource.name %></h4>
+    <p class="card-text"><%= sanitize resource.description %></p>
+    </div>
+    <div class="card-footer">
+    <a href="https://youtu.be/<%= resource.video_id %>?t=<%= resource.start_time %>" class="btn btn-primary" target='_blank'>Watch</a>
+    </div>
+  </div>
+</div>
```
Comment (garrettgsb) [inline app/views/pages/_resource-card.html.erb]: 👍

File: app/views/pages/home.html.erb [hunk 3]
```diff
@@ -39,315 +39,15 @@
...
-          </p>
-        </div>
-        <div class="card-footer">
-          <a href="https://youtu.be/NdHmGP96ow4" class="btn btn-primary" target='_blank'>Watch</a>
-        </div>
-      </div>
-    </div>
+      <% category.resources.by_date.first(3).each do |resource| %>
+        <%= render partial: 'resource-card', locals: { resource: resource } %>
+      <% end %>
```
Comment (garrettgsb) [inline app/views/pages/home.html.erb]: 👍

File: db/migrate/20200324193209_add_daily_update_videos.rb [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        ',
+        video_id: 'grnm1mR-6Pk',
+        start_time: nil,
+        date: DateTime.new(2020, 3, 19, 8, 0, 0, '-07:00'),
+        category: bc,
+        kind: 'youtube_video'
+      },
+      {
+        name: 'March 23, 2020',
+        description: '
```
Comment (garrettgsb) [inline db/migrate/20200324193209_add_daily_update_videos.rb]: This one and some others have no descriptions... Maybe that's fine, I just wanted to point it out.

File: db/seeds.rb [hunk 5]
```diff
@@ -51,15 +54,165 @@
...
+    video_id: 'NdHmGP96ow4',
+    start_time: nil,
+    date: DateTime.new(2020, 3, 19, 8, 0, 0, '-07:00'),
+    category: on,
+    kind: 'youtube_video'
   }
-  ])
+])
 
 print 'done'
```
Comment (garrettgsb) [inline db/seeds.rb]: I think of us as more of a `puts` shop but I get that you're a Python guy now :)

File: app/views/pages/_resource-card.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<div class="col-md-4 mb-5">
+  <div class="card h-100">
+    <div class="embed-responsive embed-responsive-16by9 card-img-top">
+    <iframe class="embed-responsive-item" src="https://www.youtube.com/embed/<%= resource.video_id %>?t=<%= resource.start_time %>"
```
Comment (jacquesporveau) [inline app/views/pages/_resource-card.html.erb]: Seems like we are making an assumption that we will only ever use youtube videos here. Is that safe to make?

File: app/views/pages/home.html.erb [hunk 7]
```diff
@@ -39,315 +39,15 @@
...
-        <div class="card-body">
-          <h4 class="card-title">March 20, 2020</h4>
-          <p class="card-text">
-
-          </p>
-        </div>
-        <div class="card-footer">
-          <a href="https://youtu.be/10b5fzKqfdM" class="btn btn-primary" target='_blank'>Watch</a>
-        </div>
+    <% Category.where(kind: 'daily_update').order(:priority).each do |category| %>
```
Comment (jacquesporveau) [inline app/views/pages/home.html.erb]: Seems like this `daily_update` is a special key word for our application. What do you think about defining `daily_update` as a constant on the `Category` model and maybe doing a scope for this? Seems like a significant query to be including in a view.

File: db/seeds.rb [hunk 8]
```diff
@@ -1,5 +1,8 @@
 # frozen_string_literal: true
 
+User.destroy_all
+Category.destroy_all
```
Comment (jacquesporveau) [inline db/seeds.rb]: Why do we need to do this?

File: app/models/category.rb (hunk 1 above)
Comment (jmatsuba) [reply app/models/category.rb]: ya i would go with nullify. I'll change in a min.

File: db/seeds.rb (hunk 5 above)
Comment (jmatsuba) [reply db/seeds.rb]: haha that was me... i wanted a same line seeding.......done

File: db/seeds.rb (hunk 8 above)
Comment (jmatsuba) [reply db/seeds.rb]: It would allow you to keep re-seeding your db, and not get duplicates.

Comment (jmatsuba) [reply db/seeds.rb]: However, i agree not needed. I would argue locally we should be running a `rails db:reset` and in production we would use migrations to add data critical data and a rake task for other non critical data.

File: app/views/pages/_resource-card.html.erb (hunk 6 above)
Comment (jmatsuba) [reply app/views/pages/_resource-card.html.erb]: I agree with @jacquesporveau in the resources page we have other kinds. However, to get this into prod i'll let it slide. I'll worry about it when getting the resources page working.

File: app/views/pages/home.html.erb (hunk 7 above)
Comment (jmatsuba) [reply app/views/pages/home.html.erb]: @jacquesporveau  normally i'd agree but for our hacky app its good enough for now.

File: db/migrate/20200324193209_add_daily_update_videos.rb (hunk 4 above)
Comment (NimaBoscarino) [reply db/migrate/20200324193209_add_daily_update_videos.rb]: Yeah I just stole the data that was already in the home page, and these videos didn't have a description yet.

**Pull request merged**